tools/mutation_test/mutator.py

```diff
@@ -0,0 +1,499 @@
+        """
+        for i in range(len(self.src)):
+            shutil.copyfile(self.olds[i], self.modified[i])
+
+    def cleanup(self):
+        """ Cleans up by deleting all old files
+        """
+        self.restore()
+        for i in range(len(self.src)):
+            os.remove(self.olds[i])
+
+    def mutate(self, occurrence: Occurrence) -> Tuple[str, str]:
+        """ Mutates given `occurrence` using `occurrence.pattern`
+
+        Returns the original line and the mutated line
+
+        See mutator.py for information on what a Occurrence object is
+        """
+        mutation_pattern = occurrence.pattern
+        source_code = open(occurrence.file).read().split('\n')
+        sys.stderr.write("\n==> @ Line: " + str(occurrence.line) + "\n\n")
+        sys.stderr.write("Original Line  : " + source_code[occurrence.line - 1].strip() + "\n")
+        mutated_line = (source_code[occurrence.line - 1][0:occurrence.index] + 
+            source_code[occurrence.line - 1][occurrence.index:]
+                .replace(mutation_pattern.pattern, mutation_pattern.transformation, 1))
+        sys.stderr.write("After Mutation : " + mutated_line.strip() + "\n")
+        self.write_to_file(occurrence.file.rstrip('.old'), source_code, occurrence.line - 1, mutated_line)
+        return source_code[occurrence.line - 1], mutated_line
+    
+    def write_to_file(self, path, source_code, mutated_line_number, mutated_line):
+        """ Writes a file to `path` by replacing line at `mutated_line_number` in `source_code`
+        with `mutated_line`.
+
+        Helper to separate write to file process with mutate function.
+        """
+        output_file = open(path, "w")
+        for i in range(0,len(source_code)) :
+            if i == mutated_line_number : 
+                output_file.write("/* XXX: original code was : " + source_code[i] + " */\n")
+                output_file.write(mutated_line + "\n")
+            else :
+                output_file.write(source_code[i] + "\n")
+        output_file.close()
+        sys.stderr.write("\nOutput written to " + path + "\n\n")
+
+    def findOccurrences(self, mutation_pattern: Pattern) -> List[Occurrence]:
+        """ Finds all occurrences of given `mutation_pattern` in current Mutator
+
+        Returns a list of Occurrence objects (see mutator.Occurrence in mutator.py)
+        
+        If there are multiple source files, searches all of them. Occurrence object
+        contains fields to access file, line, and index within line to find substring.
+        """
+        occurrences = []
+        m = mutation_pattern.pattern
+        for f in self.olds:
+            source_code = open(f).read().split('\n')
+            for line in self.lines_to_mutate[f]:
+                # do not mutate preprocessor, comments, or assert statements
+                if (source_code[line - 1].strip().startswith("#") or 
+                    source_code[line - 1].strip().startswith("/*") or
```

Comment:
Check for all lines between `/*` and `*/` for multiline comments

Reply:
https://github.com/aws/amazon-freertos/pull/2475/commits/07f917073798f3c82f98534c3763faf641369b14
Does this work?

Reply:
https://github.com/aws/amazon-freertos/pull/2475/commits/ad49bdec868f800f447b238ed16082ee00c4209b